parser.c3

```diff
@@ -0,0 +1,658 @@
+
+function get_next_at_identifier : (cursor : Cursor&, context : LegalEndOfToken) -> ProgramSlice = {
+    runtime_assert(cursor:try_skip_next("@"));
+    let identifier : ProgramSlice = cursor:get_next_ascii_only_identifier(context);
+    identifier.start -= 1;
+    return identifier;
+}
+
+function parse_comma_separated_named_members : (cursor : Cursor&, matched_pair : StringView) -> Vector<Argument> = {
+    let result : Vector<Argument> = make_vector<Argument>();
+
+    let is_missing_comma : bool = false;
+
+    while !cursor:try_skip_next_and_surrounding_white_space(matched_pair) {
+        if cursor:get_char() == "\0"[0] {
+            syntax_error(cursor.position, "Unterminated comma separated list");
+        }
+        if is_missing_comma {
+            syntax_error(cursor.position, "TODO: missing comma");
+        }
+
+        // Argument name
+        let name : StringView = cursor:get_next_ascii_only_identifier({comma : false, operator : false, colon : true});
+        if !cursor:try_skip_next_and_surrounding_white_space(":") {
+            syntax_error(cursor.position, "TODO: missing ':'");
+        }
+
+        // Argument type
+        let type : Type = cursor:parse_type({comma : true, operators : false, colon : false});
+        result:push_back({name, type});
+
+        // Argument separators
+        if !cursor:try_skip_next_and_surrounding_white_space(",") {
+            is_missing_comma = true;
+        }
+    }
+    return result;
+}
+
+function parse_generic_definition_list : (cursor : Cursor&) -> Vector<GenericDefinition> = {
+    runtime_assert(cursor:skip_next("["));
+    cursor:skip_natural_break_if_needed({comma: false, operators: false, colon: false});
+
+    let is_missing_comma : bool = false;
+
+    let generics : Vector<GenericDefinition> = make_vector<GenericDefinition>(cursor.allocator);
+    while cursor:try_skip_next_and_surrounding_white_space("]") {
+        if cursor:get_char() == "\0"[0] {
+            syntax_error(cursor.position, "Unterminated generic definition list");
+        }
+        if is_missing_comma {
+            syntax_error(cursor.position, "TODO: missing comma");
+        }
+
+        if cursor:get_char() == "@"[0] {
+            generics:push_back({
+                1 /* generic numeric */,
+                cursor:get_next_at_identifier({comma : true, operators : false, colon : false}),
+            });
+        } else {
+            generics:push_back({
+                0 /* generic type */,
+                cursor:get_next_ascii_only_identifier({comma : true, operators : false, colon : false}),
+            });
+        }
+
+        if !cursor:try_skip_next_and_surrounding_white_space(",") {
+            is_missing_comma = true;
+        }
+    }
+    return generics;
+}
+
+function parse_type : (cursor : Cursor&, context : LegalEndOfToken) -> Type = {
+
+}
+
+// BUG: Both `cursor` and `Cursor` are highlighed on the next line
+function parse_typedef : (cursor : Cursor&) -> Typedef = {
```

Comment:
ah lol, it's cause `parse_typedef : (cursor` looks like a typedef

Reply:
Fair enough. I'll remove the comments that I've already opened issues for